google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/cloud/AppEngineEnvironment.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright (C) 2016 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.tools.intellij.appengine.cloud;
+
+import com.google.cloud.tools.intellij.util.GctBundle;
+
+/**
+ * Specifies an App Engine environment.
+ */
+public enum AppEngineEnvironment {
+  APP_ENGINE_STANDARD("appengine.environment.name.standard"),
+  APP_ENGINE_FLEX("appengine.environment.name.flexible");
+
+  private String label;
+
+  AppEngineEnvironment(String label) {
+    this.label = label;
+  }
+
+  @Override
+  public String toString() {
```

Comment:
For me toString() returns pretty printed object state, but isn't part of
the stuff displayed in a UI unless the type is explicitly some kind of UI
element that would have text.

When I saw it called earlier, I wondered why we weren't localizing, but if
I had read environment.localizedLabel(), I wouldn't have wondered that.

Another case might be you are logging errors including environment
information.  Now when you're looking at feedback you see that the crash
was caused in an environment, but the environment is in Swahili so you're
not quite sure which one it is.q

On Wed, Jun 8, 2016 at 4:33 PM, Etan Shaul <notifications@github.com> wrote:

> In
> google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/cloud/AppEngineEnvironment.java
> <https://github.com/GoogleCloudPlatform/gcloud-intellij/pull/700#discussion_r66334181>
> :
>
> > +
> > +/**
> > + * Specifies an App Engine environment.
> > + */
> > +public enum AppEngineEnvironment {
> > +  APP_ENGINE_STANDARD("appengine.environment.name.standard"),
> > +  APP_ENGINE_FLEX("appengine.environment.name.flexible");
> > +
> > +  private String label;
> > +
> > +  AppEngineEnvironment(String label) {
> > +    this.label = label;
> > +  }
> > +
> > +  @Override
> > +  public String toString() {
>
> Really? With toString its the string representation of an
> AppEngineEnvironment which seemed pretty clear to me. It also has the
> benefit of not having to call a method for things like logging. I don't
> have strong feelings about this though.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/GoogleCloudPlatform/gcloud-intellij/pull/700/files/a3a2318d273b9729e13906620a2014cc69c7e449#r66334181>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AHf5HQ9M2sAFfcHgZSPeT_4yC4vN5Uwaks5qJyckgaJpZM4Im4aR>
> .
>